os/arch/arm/src/artik053/src/artik053_crashdump.c

```diff
@@ -0,0 +1,178 @@
+	up_puts("Disconnect this serial terminal and Run Ramdump Tool\n");
+	up_puts("****************************************************\n");
+
+	/* Receive hanshake string from HOST */
+	do {
+		host_buf[0] = up_getc();
+	} while (host_buf[0] != target_str[0]);
+
+	for (i = 1; i < strlen(target_str);) {
+		if ((ch = up_getc()) != -1) {
+			host_buf[i] = ch;
+			i++;
+		}
+	}
+
+	if (strncmp(host_buf, target_str, strlen(target_str)) != 0) {
+		/* Send NAK */
+		up_lowputc('N');
+	}
+
+	/* Send ACK */
+	up_lowputc('A');
+
+	/* Send RAM address */
+	ptr = (uint8_t *)&address;
+	for (i = 0; i < sizeof(address); i++) {
+		up_lowputc((uint8_t)*ptr);
+		ptr++;
+	}
+
+	/* Send RAM size */
+	ptr = (uint8_t *)&size;
+	for (i = 0; i < sizeof(size); i++) {
+		up_lowputc((uint8_t)*ptr);
+		ptr++;
+	}
+
+	/* Send RAMDUMP of ram_size bytes */
+	ptr = (uint8_t *)address;
+	while (size) {
+		up_lowputc((uint8_t)*ptr);
+		ptr++;
+		size--;
+	}
+
+	lldbg(" Successfull\n");
+
+	return OK;
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: board_crashdump
+ *
+ * Description:
+ *   If CONFIG_BOARD_CRASHDUMP is selected then up_asseert will call out to
+ *   board_crashdump prior to calling exit in the case of an assertion failure.
+ *   Or in the case of a hardfault looping indefinitely. board_crashdump then
+ *   has a chance to save the state of the machine. The provided
+ *   board_crashdump should save as much information as it can about the cause
+ *   of the fault and then most likely reset the system.
+ *
+ *   N.B. There are limited system resources that can be used by the provided
+ *   board_crashdump function. The tems from the fact that most critical/fatal
+ *   crashes are because of a hard fault or during interrupt processing.
+ *   In these cases, up_assert is running from the context of an interrupt
+ *   handlerand it is impossible to use any device driver in this context.
+ *
+ *   Also consider the following: Who knows what state the system is in? Is
+ *   memory trashed? Is the Heap intact? Therefore all we can expect to do in
+ *   board_crashdump is save the "machine state" in a place where on the next
+ *   reset we can write it to more sophisticated storage in a sane operating
+ *   environment.
+ *
+ ****************************************************************************/
+
+void board_crashdump(uint32_t cur_sp, void *tcb, uint8_t *filename, int lineno)
+{
+	int ret = OK;
+
+#if defined(CONFIG_S5J_WATCHDOG)
+	/* system under panic and we are dumping system state.
+	 * watchdog reset might stall the system, causing crashdump hang.
+	 * Hence let's disable watchdog reset during crashdump process.
+	 */
+	s5j_watchdog_disable();
+#endif
+
+#if defined(CONFIG_BOARD_RAMDUMP_UART)
```

Comment:
BOARD_RAMDUMP_UART need not enabled by default when BOARD_CRASHDUMP is enabled.

So i think,
config BOARD_RAMDUMP_UART
  	bool "Enable ramdump via UART"
 +	depends on BOARD_CRASHDUMP
would be enough.

Reply:
When we enable BOARD_CRASHDUMP, we should enable one among options at least. If we enable noone, BOARD_CRASHDUMP is redundant. That's why I said like above. Another method can be a default but anyway at least one of them should be enabled like debug and debug_error.

Reply:
Yeah I agree.
But how BOARD_CRASHDUMP can be "de-selected" if none is enabled (because some boards  may not support any option and we cannot restrict to enable minimum one). Please suggest

Reply:
If you want to "de-select", you should add codes in mkconfig.c.
i.e.
```
	printf("#ifdef CONFIG_DISABLE_MOUNTPOINT\n");
	printf("# undef CONFIG_FS_FAT\n");
	printf("# undef CONFIG_FS_ROMFS\n");
	printf("# undef CONFIG_FS_NXFFS\n");
	printf("# undef CONFIG_FS_SMARTFS\n");
	printf("#endif\n\n");
```